bridge/doctrine-phpcr-odm/src/FieldMapper.php

```diff
@@ -31,16 +31,16 @@ public function __construct(
         $this->fieldLoader = $fieldLoader;
     }
 
-    public function __invoke($fieldName, LoadedField $loadedField, ClassMetadata $metadata, array $extraOptions = [])
+    public function __invoke($fieldName, Field $field, ClassMetadata $metadata, array $extraOptions = [])
     {
-        $configuredType = $loadedField->getStorageType();
-        $type = $configuredType->getInnerType();
+        $refl = new \ReflectionClass($field->getStorageType());
+        $type = $refl->newInstance();
         $options = array_merge([
             'multivalue' => false,
         ], $extraOptions);
 
         if ($type instanceof ObjectType) {
```

Comment:
If somebody were to create a type that extended another, the only purpose it would serve would be add new options.

In which case they could plausibly create a custom subscriber to act upon any new options and build upon the metadata ..